packages/share/ios/Classes/FLTSharePlugin.m

```diff
@@ -69,19 +121,37 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
         return;
       }
 
-      NSNumber *originX = arguments[@"originX"];
-      NSNumber *originY = arguments[@"originY"];
-      NSNumber *originWidth = arguments[@"originWidth"];
-      NSNumber *originHeight = arguments[@"originHeight"];
+      [self shareText:shareText
+                 subject:shareSubject
+          withController:[UIApplication sharedApplication].keyWindow.rootViewController
+                atSource:originRect];
+      result(nil);
+    } else if ([@"shareFiles" isEqualToString:call.method]) {
+      NSArray *paths = arguments[@"paths"];
+      NSArray *mimeTypes = arguments[@"mimeTypes"];
+      NSString *subject = arguments[@"subject"];
+      NSString *text = arguments[@"text"];
 
-      CGRect originRect = CGRectZero;
-      if (originX != nil && originY != nil && originWidth != nil && originHeight != nil) {
-        originRect = CGRectMake([originX doubleValue], [originY doubleValue],
-                                [originWidth doubleValue], [originHeight doubleValue]);
+      if (paths.count == 0) {
+        result([FlutterError errorWithCode:@"error"
+                                   message:@"Non-empty paths expected"
+                                   details:nil]);
+        return;
       }
 
-      [self share:shareText
-                 subject:shareSubject
+      for (NSString *path in paths) {
+        if (path.length == 0) {
+          result([FlutterError errorWithCode:@"error"
+                                     message:@"Non-empty paths expected"
```

Comment:
Maybe `Each path in `paths` must not be empty`?

Reply:
Copy paste mistake 🙈 . Replaced message with `Each path must not be empty`. Felt like `each path in paths` was not adding much more info.